packages/agents/src/index.ts

```diff
@@ -759,6 +773,64 @@ export class Agent<Env, State = unknown> extends Server<Env> {
     // biome-ignore lint/complexity/noBannedTypes: <explanation>
     return callableMetadata.has(this[method as keyof this] as Function);
   }
+
+  MCPConnections: Record<string, MCPClientConnection> = {};
+
+  /**
+   * Connect to and register an MCP server
+   *
+   * @param transportConfig Transport config
+   * @param clientConfig Client config
+   * @param capabilities Client capabilities (i.e. if the client supports roots/sampling)
+   */
+  async connectToMCPServer(
+    url: URL,
+    info: ConstructorParameters<typeof Client>[0],
+    opts: {
+      transport: SSEClientTransportOptions;
+      client: ConstructorParameters<typeof Client>[1];
+      capabilities: ClientCapabilities;
+    } = { transport: {}, client: {}, capabilities: {} }
+  ) {
+    if (info.name in this.MCPConnections) {
+      throw new Error(
+        `An existing MCP client has already been registered under the name "${info.name}". The MCP client name must be unique.`
+      );
+    }
+
+    this.MCPConnections[info.name] = new MCPClientConnection(url, info, opts);
+    await this.MCPConnections[info.name].init();
+  }
+
+  /**
+   * Utility functions to get namespaced data for all MCP connections
+   */
+  listTools(): Tool[] {
+    return getNamespacedData(this.MCPConnections, "tools");
+  }
+
+  listPrompts(): Prompt[] {
+    return getNamespacedData(this.MCPConnections, "prompts");
+  }
```

Comment:
Should we include this yet? If we don't yet totally support it? Or do we and I don't see it?

Reply:
Yeah, no reason not to IMO. I'm going to go ahead and add the other prompt/resource functionality tomorrow, so it should be mostly supported if anyone wants to use it.